traits/src/aead/arrayref.rs

```diff
@@ -0,0 +1,58 @@
+//! This module contains the trait and related errors for an Authenticated
+//! Encryption with Associated Data (AEAD) scheme that takes array references
+//! as arguments and writes outputs to mutable array references.
+
+/// An Authenticated Encryption with Associated Data (AEAD) scheme. This trait
+/// is low-level and is mostly used for implementing other, more usable APIs.
+pub trait Aead<const KEY_LEN: usize, const TAG_LEN: usize, const NONCE_LEN: usize> {
+    /// Encrypt a plaintext message, producing a ciphertext and an authentication tag.
+    /// The `ciphertext` argument must have length `plaintext.len() + TAG_LEN`.
```

Comment:
Why does the tag need to go into the ciphertext? That's what the separate `tag` argument is there for? Same goes for decrypt.